modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryImpl.java

```diff
@@ -230,6 +231,11 @@ private Map<String, Object> _create(
 
 		ResourceBundle resourceBundle = getResourceBundle(locale);
 
+		JSONArray availableLocalesJSONArray =
+			ddmFormRenderingContext.getAvailableLocales();
+
+		templateContext.put("availableLocales", availableLocalesJSONArray);
+
```

Comment:
@igor-franca reorder:

```java
templateContext.put("activePage", ...)

JSONArray availableLocalesJSONArray = ..;

templateContext.put("availableLocales", availableLocalesJSONArray);

_setDDMFormFieldsEvaluableProperty(ddmForm, ddmFormLayout);

Locale locale = ddmFormRenderingContext.getLocale();

if (locale == null) {
    locale = LocaleThreadLocal.getSiteDefaultLocale();
}

ResourceBundle resourceBundle = getResourceBundle(locale);

...
```

Reply:
Just a tip: you can't insert new code between a variable's assignment and its use.

```java
ResourceBundle resourceBundle = getResourceBundle(locale);

_language.get(resourceBundle, "cancel"));
```

Reply:
Thanks for the tip Carol, I had that in mind, I think it was a lack of attention 😪

Reply:
> @igor-franca reorder:
> 
> ```java
> templateContext.put("activePage", ...)
> 
> JSONArray availableLocalesJSONArray = ..;
> 
> templateContext.put("availableLocales", availableLocalesJSONArray);
> 
> _setDDMFormFieldsEvaluableProperty(ddmForm, ddmFormLayout);
> 
> Locale locale = ddmFormRenderingContext.getLocale();
> 
> if (locale == null) {
>     locale = LocaleThreadLocal.getSiteDefaultLocale();
> }
> 
> ResourceBundle resourceBundle = getResourceBundle(locale);
> 
> ...
> ```

Thanks for the review Carol, SF requested an inline after this reordering 👍🏾